src/main.ts

```diff
@@ -1,18 +1,56 @@
 import * as core from '@actions/core'
-import {wait} from './wait'
+import {forkRepo, inviteMember, isOrgMember, isValidLicense} from './github'
+
+export async function run(): Promise<void> {
+  const owner: string = core.getInput('owner', {required: true})
+  const repo: string = core.getInput('repo', {required: true})
+  const org: string = core.getInput('org', {required: false})
+  const user: string = core.getInput('user', {required: false})
+  const addUser: boolean = core.getBooleanInput('addUser', {required: false})
+  const licenseWhitelist: string[] = core.getMultilineInput(
+    'licenseWhitelist',
+    {
+      required: false
+    }
+  )
 
-async function run(): Promise<void> {
   try {
-    const ms: string = core.getInput('milliseconds')
-    core.debug(`Waiting ${ms} milliseconds ...`) // debug is only output if you set the secret `ACTIONS_RUNNER_DEBUG` to true
+    // Optionally enforce a whitelist of allowed repository licenses for forking
+    if (!licenseWhitelist.includes('undefined')) {
+      core.info(
+        `⚖️ Checking repository license for ${repo} against provided whitelist...`
+      )
+      if (await isValidLicense(owner, repo, licenseWhitelist)) {
+        core.info(`✅ Valid license, proceeding with fork creation`)
+      } else {
+        core.setFailed(
+          `🚨 License not found in whitelist, please check to ensure the repository is compliant`
+        )
+      }
+    }
 
-    core.debug(new Date().toTimeString())
-    await wait(parseInt(ms, 10))
-    core.debug(new Date().toTimeString())
+    // Fork the specified repo into user namespace, unless an organization is specified
+    core.info(`⑂ Creating fork of repository ${repo}...`)
+    await forkRepo(owner, repo, org)
```

Comment:
What if the target already exists? Is there value in checking that before the fork operation and alerting the user?

> Cannot fork `org/repo` into `other-org/repo-fork`; `other-org/repo-fork` already exists.

Reply:
Fair point, it definitely wouldn't hurt to print out an `INFO` level message to note that a previous fork exists. We can continue to provide the URL to the forked repo so the user can go confirm for themselves that it's already there.  

I'd still want to allow `forker` to proceed after informing the user, though, since it makes real-world integration testing significantly easier than needing to come up with a new target repository and destination organization to fork each time.

Reply:
That two repositories in different places have the same name doesn’t mean they’re the same, though, right? May be wiser to bail in that case. If it’s the same repo, the user can update from upstream. If it’s a different repo, the user will need to sort that out as they see fit.

Reply:
Oh, I get what you mean now. So an edge case like this....

Years ago, https://github.com/bobsmith/magicscript was forked to https://github.com/janedoe/magicscript

Now, `janedoe` wants to fork https://github.com/abcinc/magicscript, but a repository called `magicscript` already exists in that namespace.

I'm actually not sure how Github handles that case, going to test it out now...

Reply:
Looks like the StackOverflow consensus is that Github will default to redirecting you to whichever fork was originally created: https://stackoverflow.com/questions/8290678/how-do-you-fork-a-repo-whose-upstream-youve-already-forked

I'm going to see if the forks API allows you to delineate between _true_ duplicates and unintentional duplicates (eg. where the user or org name differs from the original fork source)

Reply:
> Years ago, https://github.com/bobsmith/magicscript was forked to https://github.com/janedoe/magicscript
>
> Now, janedoe wants to fork https://github.com/abcinc/magicscript, but a repository called magicscript already exists in that namespace.

Yes, precisely this. Bear in mind, though, that the target may not be a fork. I could have a repo called `forker` that's a source repo containing the Markdown files for my blog about cutlery. Separate failure case there, I'd expect.

Reply:
All right, so I dove into the [forks API](https://docs.github.com/en/rest/reference/repos#create-a-fork) behavior and have some good and bad news:

* **Good news:** Github is smart enough to discern when a same-named repository, in fact, contains different source code. Rather than failing or doing nothing, the API instead auto-increments the repository name, so `my-forked-repo` becomes `my-forked-repo-1` if you already have a repository named `my-forked-repo` in your destination. This is something I was able to add specific handling for by parsing the last couple characters of the fork repository URL included in the response data, so when this happens, a Warning is printed so the user can investigate further, should they choose to.
* **Bad news:** Github will happily return a 202 `Accepted` response code regardless of whether you are forking a repository for the very first time, or forked it some time ago and then forgot about it. All of the other response codes offered by the API pertain to real errors, such as forking a nonexistent repository, or attempting to create a fork somewhere you don't have permission for. I also inspected the entire raw response, headers, etc. to see if there was any discernible difference between creating a brand-new fork, and requesting essentially a duplicate of an existing fork, and the responses are basically identical. 

Overall, the latter scenario feels like a relatively benign edge case that we don't necessarily need to protect the user against, short of introducing a bunch of additional overhead with extra API calls that parse the entire catalog of existing repositories on a target account/organization prior to even initiating the fork creation (which I'd prefer not to do within the scope of this PR).

Reply:
For the latter case, you could potentially check for the existence of the repo prior to trying to fork, but that's a change that can be deferred until it's needed IMO.

Reply:
Agreed!